src/test-error/index.js

```diff
@@ -0,0 +1,40 @@
+import { parser } from 'sax';
+import TEMPLATES from './templates';
+
+export default function format (err, decorator, viewportWidth, cb) {
+    var msg       = TEMPLATES[err.code](err, viewportWidth);
+    var xmlParser = parser(false, { lowercase: true });
+    var tagsStack = [];
+
+    xmlParser.onerror = function (e) {
+        throw e;
+    };
+
+    xmlParser.onopentag = function (node) {
+        node.content = '';
+        tagsStack.push(node);
+    };
+
+    xmlParser.ontext = function (txt) {
+        tagsStack[tagsStack.length - 1].content += txt;
+    };
+
+    xmlParser.onclosetag = function (tag) {
+        var topNode = tagsStack[tagsStack.length - 1];
+
+        if (tag === topNode.name && tagsStack.length > 1) {
+            var type     = topNode.attributes['data-type'];
+            var selector = type ? `${tag} ${type}` : `${tag}`;
+
+            tagsStack.pop();
+            tagsStack[tagsStack.length - 1].content += decorator[selector](topNode.content, topNode.attributes);
+        }
+    };
+
+    xmlParser.onend = function () {
+        cb(tagsStack[0].content);
```

Comment:
It's will not matter if we will move to the parse5, but why callback-approach here? Why not promise?